site/docs/main/configuration.md

```diff
@@ -1,9 +1,60 @@
 # Contour Configuration Reference
 
+- [Serve Flags](#serve-flags)
 - [Configuration File](#configuration-file)
 - [Environment Variables](#environment-variables)
 - [Bootstrap Config File](#bootstrap-config-file)
 
+## Overview
+
+There are various ways to configure Contour, flags, the configuration file, as well as environment variables.
+Contour has a precedence of configuration for contour serve, meaning anything configured in the config file is overridden by environment vars which are overridden by cli flags.
+
+## Serve Flags
+
+The `contour serve` command is the main command which is used to watch for Kubernetes resource and process them into Envoy configuration which is then streamed to any Envoy via its xDS gRPC connection.
+There are a number of flags that can be passed to this command which further configures how Contour operates. 
+Many of these flags are mirrored in the [Contour Configuration File](#configuration-file).
+
+| Flag Name         | Description        |
+|-------------------|--------------------|
+| --config-path       | Path to base configuration |
+| --incluster         | Use in cluster configuration |
+| --kubeconfig=KUBECONFIG |    Path to kubeconfig (if not in running inside a cluster) |
+| --xds-address=XDS-ADDRESS | xDS gRPC API address |
+| --xds-port=XDS-PORT       | xDS gRPC API port |
+| --stats-address=STATS-ADDRESS | Envoy /stats interface address |
+| --stats-port=STATS-PORT  |  Envoy /stats interface port |
+| --debug-http-address=DEBUG-HTTP-ADDRESS | Address the debug http endpoint will bind to. |
+| --debug-http-port=DEBUG-HTTP-PORT  | Port the debug http endpoint will bind to |
+| --http-address=HTTP-ADDRESS  | Address the metrics HTTP endpoint will bind to |
+| --http-port=HTTP-PORT  |    Port the metrics HTTP endpoint will bind to. |
+| --health-address=HEALTH-ADDRESS |   Address the health HTTP endpoint will bind to |
+| --health-port=HEALTH-PORT | Port the health HTTP endpoint will bind to |
+| --contour-cafile=CONTOUR-CAFILE | CA bundle file name for serving gRPC with TLS |
```

Comment:
If we change the format as I suggested for other things, we should reflect that these can be environment variables:
```suggestion
| --contour-cafile=<string|CONTOUR_CAFILE> | CA bundle file name for serving gRPC with TLS |
```
Or we could add another column "Environment variable" also.